dojo/models.py

```diff
@@ -998,6 +1038,37 @@ class Product(models.Model):
         blank=False,
         verbose_name=_("Disable SLA breach notifications"),
         help_text=_("Disable SLA breach notifications if configured in the global settings"))
+    async_updating = models.BooleanField(default=False,
+                                            help_text=_('Findings under this SLA configuration are asynchronously being updated'))
+
+    def save(self, *args, **kwargs):
+        # get the product's sla config before saving (if this is an existing product)
+        initial_sla_config = None
+        if self.pk is not None:
+            initial_sla_config = getattr(Product.objects.get(pk=self.pk), 'sla_configuration', None)
+            # if initial sla config exists and async finding update is already running, revert sla config before saving
+            if initial_sla_config and self.async_updating:
+                self.sla_configuration = initial_sla_config
+
+        super(Product, self).save(*args, **kwargs)
+
+        # if the initial sla config exists and async finding update is not running
+        if initial_sla_config is not None and not self.async_updating:
+            # get the new sla config from the saved product
+            new_sla_config = getattr(self, 'sla_configuration', None)
+            # if the sla config has changed, update finding sla expiration dates within this product
+            if new_sla_config and (initial_sla_config != new_sla_config):
```

Comment:
Won't this be true in both the case when the same SLA config has been modified and the case where a new SLA config has been selected? Seems like the SLA config should be marked `async_updating` only in the first case, not the second?

Reply:
We have to block both directions. For example, if `SLA_Configuration A` is used by `Product A`, but a new `SLA_Configuration B` is applied to `Product A`, we need to lock `SLA_Configuration B` from being changed (while the asynchronous update occurs) _and_ `Product A` from changing to a different SLA configuration (also while the asynchronous update occurs).

Reply:
This is also true for the other way around. If `Product A` is using `SLA_Configuration A`, and someone changes `SLA_Configuration A`'s SLA days, we need to block all products utilizing `SLA_Configuration A` from changing to a different SLA configuration while that async task runs. The reason I blocked this is we have no way to cancel a celery task once it has started (I could not find a way, at least).